### DISCUSSION TITLE
Cannot place slider nodes outside the playfield area in editor

### DISCUSSION THREAD

**Discussion by gustarmartins** [ppy/osu#16572] [Q&A] [answered]
When you right click a slider to finish creating it, and is off the grid, it won't finish. 

https://mkguh.s-ul.eu/k5kPtPLE.mkv - video showing the issue

**Comment by frenzibyte** [comment]
Appears to be something osu!(stable) supports, but not sure how simple that would be for osu!(lazer) to support at all.

What you can currently do in osu!(lazer) is to finish the slider in bounds and then move the final node wherever you want outside the bounds.

**Comment by peppy** [comment]
With the new curve creation options, it should be less of an issue. Have you tried using the per-segment curve settings? They should let you create sliders that would have previously required silly out-of-bounds placements without much effort.

**Reply by gustarmartins** [reply]
This is not practical for when mapping Easy/Normal difficulties where most mappers are used to the distance snapping tool, which will often make you have to finish sliders a little bit outside of the grid.

I absolutely love the new curve creation options, but I'm afraid it's not what my issue is originally related about. (Yes, I understand you meant that mappers don't need to add lots of slider nodes off the grid now, but in this case it's more related to low diff mapping, I suppose)

I haven't tested for Hard/Insane mapping yet.

**Reply by gustarmartins** [reply]
I think the video I posted express well what this is about.

![image](https://user-images.githubusercontent.com/45127556/150833805-ad0ab3a9-619b-4c0f-8c0f-278838ec17e4.png)

This pattern on an Easy difficulty for example is something I did using distance snap and blankets, and the (4) would originally have to have its slider end outside of the grid to fit my pattern design. For that I had to move the slider inside the grid to finish creating it, then move it back. If anything, this is a minor annoyance, maybe not a bug exactly.

**Reply by peppy** [reply]
Is that not a breach of the ranking criteria..?

If that's allowed, we should instead be discussing adjusting the playfield width.

**Reply by Gamingzone123** [reply]
the ranking criteria states that it must to go outside a 4:3 aspect ratio, but the grid doesn't actually represent that, at least not in stable to my experience

**Reply by gustarmartins** [reply]
No, @peppy it is not. What is against the ranking criteria is having offscreen objects in the gameplay of at least 4:3 aspect ratios. As said by Gamingzone123, the grid in stable never really represented that. (As in objects off the grid are not necessarily offscreen in the gameplay) and mappers are aware of that and have been placing objects off the grid since forever. :O

**Reply by peppy** [reply]
We should fix this by expanding the playfield itself then, either with an "overflow" display in the editor or verbatim on the playfield dimensions themselves.

**Reply by apollo-dw** [reply]
Would "expanding the playfield" include moving away from a 4:3 playfield to a wider aspect ratio, or is that still a priority?

**Reply by bdach** [reply]
changing the playfield's aspect ratio sounds like a pretty brutal change to execute in a backwards compatible manner, all things considered. probably not going to be something that happens in the near future

**Comment by peppy** [comment, marked as answer]
Tracking via https://github.com/ppy/osu/issues/16737.
